In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

### Analysing Meal Info

In [2]:
meal_info = pd.read_csv("meal_info.csv")
meal_info.head()

meal_id   category cuisine
0     1885  Beverages    Thai
1     1993  Beverages    Thai
2     2539  Beverages    Thai
3     1248  Beverages  Indian
4     2631  Beverages  Indian

In [3]:
meal_info.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51 entries, 0 to 50
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   meal_id   51 non-null     int64 
 1   category  51 non-null     object
 2   cuisine   51 non-null     object
dtypes: int64(1), object(2)
memory usage: 1.3+ KB


In [4]:
meal_info.category.value_counts()

Beverages       12
Sandwich         3
Rice Bowl        3
Salad            3
Extras           3
Fish             3
Soup             3
Starters         3
Biryani          3
Pasta            3
Desert           3
Pizza            3
Other Snacks     3
Seafood          3
Name: category, dtype: int64

In [5]:
meal_info.cuisine.value_counts()

Thai           15
Indian         12
Italian        12
Continental    12
Name: cuisine, dtype: int64

###### Nothing to clean in the meal_info

### Analysing Fulfilment Centre Info

In [6]:
fulfilment_center_info = pd.read_csv("fulfilment_center_info.csv")
fulfilment_center_info.head()

center_id  city_code  region_code center_type  op_area
0         11        679           56      TYPE_A      3.7
1         13        590           56      TYPE_B      6.7
2        124        590           56      TYPE_C      4.0
3         66        648           34      TYPE_A      4.1
4         94        632           34      TYPE_C      3.6

In [7]:
fulfilment_center_info.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 77 entries, 0 to 76
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   center_id    77 non-null     int64  
 1   city_code    77 non-null     int64  
 2   region_code  77 non-null     int64  
 3   center_type  77 non-null     object 
 4   op_area      77 non-null     float64
dtypes: float64(1), int64(3), object(1)
memory usage: 3.1+ KB


In [8]:
fulfilment_center_info.city_code.value_counts()

590    9
526    8
638    3
593    2
517    2
647    2
522    2
676    2
576    2
699    2
604    2
614    2
602    1
541    1
556    1
683    1
553    1
680    1
679    1
485    1
675    1
658    1
659    1
654    1
651    1
615    1
649    1
648    1
620    1
628    1
685    1
561    1
473    1
577    1
599    1
596    1
478    1
461    1
713    1
456    1
579    1
609    1
562    1
703    1
702    1
700    1
632    1
698    1
695    1
693    1
515    1
Name: city_code, dtype: int64

In [9]:
fulfilment_center_info.region_code.value_counts()

56    30
34    21
77    17
85     5
93     1
71     1
35     1
23     1
Name: region_code, dtype: int64

In [10]:
fulfilment_center_info.center_type.value_counts()

TYPE_A    43
TYPE_C    19
TYPE_B    15
Name: center_type, dtype: int64

In [11]:
fulfilment_center_info.op_area.describe()

count    77.000000
mean      3.985714
std       1.106406
min       0.900000
25%       3.500000
50%       3.900000
75%       4.400000
max       7.000000
Name: op_area, dtype: float64

###### Nothing to clean in the meal_info

### Analysing Training Data

In [12]:
training_data = pd.read_csv("train.csv")
training_data.head()

id  week  center_id  meal_id  checkout_price  base_price  \
0  1379560     1         55     1885          136.83      152.29   
1  1466964     1         55     1993          136.83      135.83   
2  1346989     1         55     2539          134.86      135.86   
3  1338232     1         55     2139          339.50      437.53   
4  1448490     1         55     2631          243.50      242.50   

   emailer_for_promotion  homepage_featured  num_orders  
0                      0                  0         177  
1                      0                  0         270  
2                      0                  0         189  
3                      0                  0          54  
4                      0                  0          40

In [13]:
training_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 456548 entries, 0 to 456547
Data columns (total 9 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   id                     456548 non-null  int64  
 1   week                   456548 non-null  int64  
 2   center_id              456548 non-null  int64  
 3   meal_id                456548 non-null  int64  
 4   checkout_price         456548 non-null  float64
 5   base_price             456548 non-null  float64
 6   emailer_for_promotion  456548 non-null  int64  
 7   homepage_featured      456548 non-null  int64  
 8   num_orders             456548 non-null  int64  
dtypes: float64(2), int64(7)
memory usage: 31.3 MB


In [14]:
training_data.week.value_counts()

122    3359
105    3348
106    3347
140    3332
123    3331
       ... 
13     2838
12     2831
14     2830
7      2795
8      2786
Name: week, Length: 145, dtype: int64

Training data have unequal observations for weeks. That means, data has missing observations for many weeks.

In [15]:
training_data.center_id.value_counts()

13     7046
10     7015
52     6993
43     6970
67     6915
       ... 
139    4627
57     4501
162    4366
41     4083
91     3432
Name: center_id, Length: 77, dtype: int64

In [16]:
training_data.meal_id.value_counts()

2290    11138
1062    11137
1727    11136
1109    11134
1754    11132
2707    11123
1778    11121
1993    11116
1962    11114
1971    11102
1885    11092
2581    11072
2826    11057
2322    11002
1558    10952
1207    10806
2640    10747
1230    10746
1878    10527
2631    10458
2539    10451
1543    10236
2760    10209
1248     9939
2664     9853
2704     9811
1445     9727
2126     9705
1216     9695
2304     9676
2569     9459
2492     9382
1525     9359
2490     9247
2444     9097
2139     8899
2577     8667
2306     8294
2867     8092
1247     7184
1902     6748
1770     6682
1311     4682
1803     4674
1571     4511
1438     4385
2494     4206
1198     4206
1847     4084
2956     3319
2104     2357
Name: meal_id, dtype: int64

In [17]:
training_data.checkout_price.describe()

count    456548.000000
mean        332.238933
std         152.939723
min           2.970000
25%         228.950000
50%         296.820000
75%         445.230000
max         866.270000
Name: checkout_price, dtype: float64

In [18]:
training_data.base_price.describe()

count    456548.000000
mean        354.156627
std         160.715914
min          55.350000
25%         243.500000
50%         310.460000
75%         458.870000
max         866.270000
Name: base_price, dtype: float64

There is lots of difference in 25% and minimum value in checkout price. 

In [19]:
training_data.sort_values('checkout_price').head()

id  week  center_id  meal_id  checkout_price  base_price  \
110286  1300283    38         73     1230            2.97      316.25   
44875   1381921    16         41     1248           45.62      253.23   
44874   1414392    16         41     2631           47.59      253.23   
417180  1372230   134         24     2631           53.41       86.39   
72701   1324508    26        124     1803           55.35      212.43   

        emailer_for_promotion  homepage_featured  num_orders  
110286                      0                  0          14  
44875                       0                  0         123  
44874                       0                  0         163  
417180                      1                  0         134  
72701                       0                  0          81

There is an outlier with checkout_price as 2.97.

In [20]:
training_data.num_orders.describe()

count    456548.000000
mean        261.872760
std         395.922798
min          13.000000
25%          54.000000
50%         136.000000
75%         324.000000
max       24299.000000
Name: num_orders, dtype: float64

There is lots of difference in 75% and maximum value.

In [21]:
training_data.sort_values('num_orders').tail()

id  week  center_id  meal_id  checkout_price  base_price  \
413351  1346216   132         43     1971          204.70      324.04   
13923   1470267     5         10     2290          162.99      309.43   
16904   1087163     6         43     2290          156.20      310.43   
19700   1300512     7         43     2290          159.08      311.43   
14050   1452474     5         43     2290          159.08      311.43   

        emailer_for_promotion  homepage_featured  num_orders  
413351                      1                  1       13150  
13923                       1                  0       13580  
16904                       0                  1       14229  
19700                       1                  1       15336  
14050                       1                  1       24299

Another outlier is present with 24299 number of orders.

In [22]:
training_data.sort_values('num_orders').head()

id  week  center_id  meal_id  checkout_price  base_price  \
107078  1117026    37         17     1247          445.23      445.23   
432865  1322219   138        143     1230          367.63      367.63   
432866  1312985   138        143     2322          369.63      369.63   
28824   1448126    11        109     1770          484.03      486.03   
28823   1066718    11        109     1247          445.23      446.23   

        emailer_for_promotion  homepage_featured  num_orders  
107078                      0                  0          13  
432865                      0                  0          13  
432866                      0                  0          13  
28824                       0                  0          13  
28823                       0                  0          13

In [23]:
training_data.emailer_for_promotion.value_counts()

0    419498
1     37050
Name: emailer_for_promotion, dtype: int64

Data analysing summary of training data:
1. There are many missing observations for some weeks
2. There is an outlier with checkout_price as 2.97
3. Another outlier is present with 24299 number of orders

In [24]:
training_data.num_orders.describe()

count    456548.000000
mean        261.872760
std         395.922798
min          13.000000
25%          54.000000
50%         136.000000
75%         324.000000
max       24299.000000
Name: num_orders, dtype: float64

### Analysing Test Data

In [25]:
test_data = pd.read_csv("test.csv")
test_data.head()

id  week  center_id  meal_id  checkout_price  base_price  \
0  1028232   146         55     1885          158.11      159.11   
1  1127204   146         55     1993          160.11      159.11   
2  1212707   146         55     2539          157.14      159.14   
3  1082698   146         55     2631          162.02      162.02   
4  1400926   146         55     1248          163.93      163.93   

   emailer_for_promotion  homepage_featured  
0                      0                  0  
1                      0                  0  
2                      0                  0  
3                      0                  0  
4                      0                  0

In [26]:
test_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32573 entries, 0 to 32572
Data columns (total 8 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   id                     32573 non-null  int64  
 1   week                   32573 non-null  int64  
 2   center_id              32573 non-null  int64  
 3   meal_id                32573 non-null  int64  
 4   checkout_price         32573 non-null  float64
 5   base_price             32573 non-null  float64
 6   emailer_for_promotion  32573 non-null  int64  
 7   homepage_featured      32573 non-null  int64  
dtypes: float64(2), int64(6)
memory usage: 2.0 MB


In [27]:
test_data.week.value_counts()

148    3332
149    3313
150    3277
147    3271
153    3268
152    3242
146    3240
154    3228
151    3224
155    3178
Name: week, dtype: int64

Test data too have missing observations for weeks.


In [28]:
test_data.meal_id.value_counts()

1727    770
2826    770
2290    770
2664    770
1754    770
1962    770
2569    770
2581    769
1885    769
1109    769
2490    769
1971    768
1558    767
1993    767
1062    766
2707    764
1778    762
2539    761
1543    760
1207    758
2322    752
1230    744
2640    735
2704    724
2631    717
2304    715
2760    712
1216    709
1878    709
2139    671
1525    668
1248    660
2492    638
2126    626
1445    602
2306    592
2444    568
1571    544
2577    540
2867    527
1770    486
1902    435
2956    426
1247    416
2104    361
1311    342
1803    341
1847    333
1438    327
1198    309
2494    305
Name: meal_id, dtype: int64

In [29]:
test_data.center_id.value_counts()

13     509
10     508
52     506
174    504
43     499
      ... 
139    321
57     311
162    310
41     299
91     280
Name: center_id, Length: 77, dtype: int64

In [30]:
test_data.checkout_price.describe()

count    32573.000000
mean       341.854440
std        153.893886
min         67.900000
25%        214.430000
50%        320.130000
75%        446.230000
max       1113.620000
Name: checkout_price, dtype: float64

In [31]:
test_data.base_price.describe()

count    32573.000000
mean       356.493615
std        155.150101
min         89.240000
25%        243.500000
50%        321.130000
75%        455.930000
max       1112.620000
Name: base_price, dtype: float64

In [32]:
test_data.emailer_for_promotion.value_counts()

0    30409
1     2164
Name: emailer_for_promotion, dtype: int64

In [33]:
test_data.homepage_featured.value_counts()

0    29923
1     2650
Name: homepage_featured, dtype: int64

Test data also contains missing observation for weeks

### Data Preprocessing

In [34]:
#Combining Test & Train Dataset
training_data_without_target = training_data[training_data.columns[training_data.columns != 'num_orders'].values]
total_data = training_data_without_target.append(test_data, sort=False)
total_data.tail()

id  week  center_id  meal_id  checkout_price  base_price  \
32568  1250239   155         61     1543          482.09      484.09   
32569  1039516   155         61     2304          483.09      483.09   
32570  1158107   155         61     2664          322.07      323.07   
32571  1444235   155         61     2569          322.07      323.07   
32572  1291286   155         61     2490          276.45      276.45   

       emailer_for_promotion  homepage_featured  
32568                      0                  0  
32569                      0                  0  
32570                      0                  0  
32571                      0                  0  
32572                      0                  0

In [35]:
#merge fulfilment center information into total data
total_data = total_data.merge(fulfilment_center_info, on='center_id', how='left')
total_data.head()

id  week  center_id  meal_id  checkout_price  base_price  \
0  1379560     1         55     1885          136.83      152.29   
1  1466964     1         55     1993          136.83      135.83   
2  1346989     1         55     2539          134.86      135.86   
3  1338232     1         55     2139          339.50      437.53   
4  1448490     1         55     2631          243.50      242.50   

   emailer_for_promotion  homepage_featured  city_code  region_code  \
0                      0                  0        647           56   
1                      0                  0        647           56   
2                      0                  0        647           56   
3                      0                  0        647           56   
4                      0                  0        647           56   

  center_type  op_area  
0      TYPE_C      2.0  
1      TYPE_C      2.0  
2      TYPE_C      2.0  
3      TYPE_C      2.0  
4      TYPE_C      2.0

In [36]:
#merge meal info into total data
total_data = total_data.merge(meal_info, on='meal_id', how='left')
total_data.head()

id  week  center_id  meal_id  checkout_price  base_price  \
0  1379560     1         55     1885          136.83      152.29   
1  1466964     1         55     1993          136.83      135.83   
2  1346989     1         55     2539          134.86      135.86   
3  1338232     1         55     2139          339.50      437.53   
4  1448490     1         55     2631          243.50      242.50   

   emailer_for_promotion  homepage_featured  city_code  region_code  \
0                      0                  0        647           56   
1                      0                  0        647           56   
2                      0                  0        647           56   
3                      0                  0        647           56   
4                      0                  0        647           56   

  center_type  op_area   category cuisine  
0      TYPE_C      2.0  Beverages    Thai  
1      TYPE_C      2.0  Beverages    Thai  
2      TYPE_C      2.0  Beverages    Thai  
3      TYPE_C      2.0  Beverages  Indian  
4      TYPE_C      2.0  Beverages  Indian

In [37]:
total_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 489121 entries, 0 to 489120
Data columns (total 14 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   id                     489121 non-null  int64  
 1   week                   489121 non-null  int64  
 2   center_id              489121 non-null  int64  
 3   meal_id                489121 non-null  int64  
 4   checkout_price         489121 non-null  float64
 5   base_price             489121 non-null  float64
 6   emailer_for_promotion  489121 non-null  int64  
 7   homepage_featured      489121 non-null  int64  
 8   city_code              489121 non-null  int64  
 9   region_code            489121 non-null  int64  
 10  center_type            489121 non-null  object 
 11  op_area                489121 non-null  float64
 12  category               489121 non-null  object 
 13  cuisine                489121 non-null  object 
dtypes: float64(3), int64(8), object(3)
m

In [38]:
#derive mean_base_price and discount (in percentage) using checkout_price and base_price
meal_base_price = total_data[['week', 'center_id', 'meal_id', 'base_price']]
meal_base_price = meal_base_price.set_index(['meal_id', 'center_id', 'week'])
meal_base_price = meal_base_price.sort_index()
meal_base_price.head()

base_price
meal_id center_id week            
1062    10        1         181.39
                  2         182.36
                  3         182.36
                  4         183.36
                  5         181.39

In [39]:
#derive average orders
meal_per_center = training_data[['week', 'center_id', 'meal_id', 'num_orders']]
meal_per_center = meal_per_center.set_index(['meal_id', 'center_id', 'week'])
meal_per_center = meal_per_center.sort_index()
meal_per_center.head()

num_orders
meal_id center_id week            
1062    10        1            865
                  2            782
                  3            851
                  4           1202
                  5            958

In [40]:
meal_across_center = training_data[['week', 'meal_id', 'num_orders']]
meal_across_center = meal_across_center.set_index(['meal_id', 'week'])
meal_across_center = meal_across_center.sort_index()
meal_across_center.head()

num_orders
meal_id week            
1062    1            391
        1            514
        1            798
        1            284
        1            256

In [42]:
def avg_orders(row):
    if (row.meal_id, row.center_id) in meal_base_price.index:
        row['mean_base_price'] = meal_base_price.loc[(row.meal_id, row.center_id)].loc[:row.week].base_price.mean()
    else:
        row['mean_base_price'] = row['base_price']
        
    week_adj = row.week - 10
    if (row.meal_id, row.center_id) in meal_per_center.index:
        hist = meal_per_center.loc[(row.meal_id, row.center_id)]
        row['avg_orders_13week'] = hist.loc[(row.week-13):row.week].num_orders.mean()
        row['avg_orders_26week'] = hist.loc[(row.week-26):row.week].num_orders.mean()
        row['avg_orders_52week'] = hist.loc[(row.week-52):row.week].num_orders.mean()
        row['avg_orders_13week_adj'] = hist.loc[(week_adj-13):week_adj].num_orders.mean()
        row['avg_orders_26week_adj'] = hist.loc[(week_adj-26):week_adj].num_orders.mean()
    else:
        row['avg_orders_13week'] = 0
        row['avg_orders_26week'] = 0
        row['avg_orders_52week'] = 0
        row['avg_orders_13week_adj'] = 0
        row['avg_orders_26week_adj'] = 0
        
    if row.meal_id in meal_across_center.index:
        hist_across = meal_across_center.loc[row.meal_id]
        row['avg_orders_13week_across'] = hist_across.loc[(row.week-13):row.week].num_orders.mean()
        row['avg_orders_26week_across'] = hist_across.loc[(row.week-26):row.week].num_orders.mean()
        row['avg_orders_52week_across'] = hist_across.loc[(row.week-52):row.week].num_orders.mean()
        row['avg_orders_13week_adj_across'] = hist_across.loc[(week_adj-13):week_adj].num_orders.mean()
        row['avg_orders_26week_adj_across'] = hist_across.loc[(week_adj-26):week_adj].num_orders.mean()
    else:
        row['avg_orders_13week_across'] = 0
        row['avg_orders_26week_across'] = 0
        row['avg_orders_52week_across'] = 0
        row['avg_orders_13week_adj_across'] = 0
        row['avg_orders_26week_adj_across'] = 0
    
    return row
    
total_data = total_data.apply(avg_orders, axis=1)
total_data['avg_orders_13week'].describe()

count    489088.000000
mean        261.401847
std         305.371162
min           0.000000
25%          66.714286
50%         156.500000
75%         341.785714
max        8312.000000
Name: avg_orders_13week, dtype: float64

In [44]:
total_data['avg_orders_13week_adj'].describe()

count    454948.000000
mean        264.084699
std         308.607584
min           0.000000
25%          67.000000
50%         158.142857
75%         345.714286
max        8312.000000
Name: avg_orders_13week_adj, dtype: float64

In [45]:
total_data['avg_orders_13week_across'].describe()

count    489121.000000
mean        261.546329
std         228.685112
min          13.000000
25%          87.067308
50%         182.153135
75%         380.385475
max        2439.792910
Name: avg_orders_13week_across, dtype: float64

In [46]:
total_data['avg_orders_13week_adj_across'].describe()

count    459359.000000
mean        262.677119
std         231.378155
min          13.000000
25%          85.612698
50%         182.615616
75%         381.244989
max        2439.792910
Name: avg_orders_13week_adj_across, dtype: float64

In [47]:
total_data['discount'] = total_data['mean_base_price'] - total_data['checkout_price']
total_data['discount'] = total_data['discount'] / total_data['mean_base_price']
total_data.discount.describe()

count    489121.000000
mean          0.057960
std           0.130148
min          -2.239083
25%          -0.020159
50%           0.007258
75%           0.132584
max           0.990654
Name: discount, dtype: float64

In [48]:
total_data['year'] = (((total_data['week'] - 1)/52) + 1).astype('int')
total_data.year.value_counts()

3    167103
2    167029
1    154989
Name: year, dtype: int64

In [57]:
total_data['week_in_month'] = (((total_data['week'] - 1) % 4) + 1)
total_data.week_in_month.value_counts()

1    123367
2    122940
3    122788
4    120026
Name: week_in_month, dtype: int64

In [58]:

total_data['month'] = (((total_data['week'] - 1)/4).astype('int') % 13) + 1
total_data.month.value_counts()

11    38625
12    38474
8     38345
9     38278
10    38242
7     38040
5     37940
6     37724
1     37390
4     37270
3     37011
2     36849
13    34933
Name: month, dtype: int64

In [59]:
total_data['quarter'] = (((total_data['week'] - 1)/13).astype('int') % 4) + 1
total_data.quarter.value_counts()

3    124373
2    122513
4    121634
1    120601
Name: quarter, dtype: int64

In [60]:
training_data = training_data[['id', 'num_orders']].merge(total_data, on='id', how='left')
training_data.head()

id  num_orders  week  center_id  meal_id  checkout_price  base_price  \
0  1379560         177     1         55     1885          136.83      152.29   
1  1466964         270     1         55     1993          136.83      135.83   
2  1346989         189     1         55     2539          134.86      135.86   
3  1338232          54     1         55     2139          339.50      437.53   
4  1448490          40     1         55     2631          243.50      242.50   

   emailer_for_promotion  homepage_featured  city_code  ...  \
0                      0                  0        647  ...   
1                      0                  0        647  ...   
2                      0                  0        647  ...   
3                      0                  0        647  ...   
4                      0                  0        647  ...   

   avg_orders_13week_across avg_orders_26week_across  \
0                726.506494               726.506494   
1                793.896104               793.896104   
2                272.116883               272.116883   
3                 49.709677                49.709677   
4                 74.014085                74.014085   

   avg_orders_52week_across avg_orders_13week_adj_across  \
0                726.506494                          NaN   
1                793.896104                          NaN   
2                272.116883                          NaN   
3                 49.709677                          NaN   
4                 74.014085                          NaN   

  avg_orders_26week_adj_across  discount  year  month  quarter  week_in_month  
0                          NaN  0.101517     1      1        1              1  
1                          NaN -0.007362     1      1        1              1  
2                          NaN  0.007361     1      1        1              1  
3                          NaN  0.224053     1      1        1              1  
4                          NaN -0.004124     1      1        1              1  

[5 rows x 31 columns]

In [61]:
training_data.info()


<class 'pandas.core.frame.DataFrame'>
Int64Index: 456548 entries, 0 to 456547
Data columns (total 31 columns):
 #   Column                        Non-Null Count   Dtype  
---  ------                        --------------   -----  
 0   id                            456548 non-null  int64  
 1   num_orders                    456548 non-null  int64  
 2   week                          456548 non-null  int64  
 3   center_id                     456548 non-null  int64  
 4   meal_id                       456548 non-null  int64  
 5   checkout_price                456548 non-null  float64
 6   base_price                    456548 non-null  float64
 7   emailer_for_promotion         456548 non-null  int64  
 8   homepage_featured             456548 non-null  int64  
 9   city_code                     456548 non-null  int64  
 10  region_code                   456548 non-null  int64  
 11  center_type                   456548 non-null  object 
 12  op_area                       456548 non-nul

In [62]:
training_data.to_csv('train_feature.csv', index=False)


In [63]:
test_data = test_data[['id']].merge(total_data, on='id', how='left')
test_data.head()


id  week  center_id  meal_id  checkout_price  base_price  \
0  1028232   146         55     1885          158.11      159.11   
1  1127204   146         55     1993          160.11      159.11   
2  1212707   146         55     2539          157.14      159.14   
3  1082698   146         55     2631          162.02      162.02   
4  1400926   146         55     1248          163.93      163.93   

   emailer_for_promotion  homepage_featured  city_code  region_code  ...  \
0                      0                  0        647           56  ...   
1                      0                  0        647           56  ...   
2                      0                  0        647           56  ...   
3                      0                  0        647           56  ...   
4                      0                  0        647           56  ...   

  avg_orders_13week_across  avg_orders_26week_across avg_orders_52week_across  \
0               718.100000                719.609915               750.125657   
1               486.656344                493.422289               528.837337   
2               222.087613                226.483723               264.149211   
3               137.327331                116.760939               106.818255   
4                96.961538                 84.167702                72.140451   

  avg_orders_13week_adj_across  avg_orders_26week_adj_across  discount  year  \
0                   718.923721                    711.837838 -0.083689     3   
1                   485.195915                    507.465476 -0.097471     3   
2                   223.408144                    226.485104 -0.074594     3   
3                   112.733799                    110.001028  0.046584     3   
4                    81.720856                     76.970362  0.042897     3   

   month  quarter  week_in_month  
0     11        4              2  
1     11        4              2  
2     11        4              2  
3     11        4              2  
4     11        4              2  

[5 rows x 30 columns]

In [64]:
test_data.info()


<class 'pandas.core.frame.DataFrame'>
Int64Index: 32573 entries, 0 to 32572
Data columns (total 30 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   id                            32573 non-null  int64  
 1   week                          32573 non-null  int64  
 2   center_id                     32573 non-null  int64  
 3   meal_id                       32573 non-null  int64  
 4   checkout_price                32573 non-null  float64
 5   base_price                    32573 non-null  float64
 6   emailer_for_promotion         32573 non-null  int64  
 7   homepage_featured             32573 non-null  int64  
 8   city_code                     32573 non-null  int64  
 9   region_code                   32573 non-null  int64  
 10  center_type                   32573 non-null  object 
 11  op_area                       32573 non-null  float64
 12  category                      32573 non-null  object 
 13  c

In [65]:
test_data.to_csv('test_feature.csv', index=False)
